Content.Server/Nutrition/EntitySystems/DrinkSystem.cs

```diff
@@ -123,51 +126,42 @@ public float TotalHydration(EntityUid uid, DrinkComponent? comp = null)
 
     private void OnExamined(EntityUid uid, DrinkComponent component, ExaminedEvent args)
     {
-        if (!component.Opened || !args.IsInDetailsRange || !component.Examinable)
+        var canOpen = TryComp<OpenableComponent>(uid, out var openable);
+        if (_openable.IsClosed(uid, openable) || !args.IsInDetailsRange || !component.Examinable)
```

Comment:
you dont use openable again in this method so i think making canOpen equal these conditions and not doing the TryComp here would be better

Reply:
openable is not directly used, the canOpen is though. if not for the IsClosed check, it would just be a HasComp but its resolving in IsClosed so may as well use it

Reply:
i meant that its unnecessary 'cuz if you make `canOpen` equal the `IsClosed` method (may as well equal the args.IsInDetailRange too), that will handle the resolve by itself (if it doesnt have the component it will return false)

Reply:
yes but if it does have the component and is closed it will also return false, the point is that canOpen is just if it has the component